New note show template view

nicoleng12 opened the pull request:
This PR addresses jira 1091. When the user clicks on "New Note" button, the templates view is displayed in the note assistant as opposed to the shortcuts view. 

This fix required passing a property to keep track of when to show the templates view and to make sure shortcuts view is still displayed in other use cases.

In addition, PropTypes were fixed in the files I touched and Full App UI test was updated to handle the new workflow when clicking "New Note" button.

Cases to test:
- On app start, click "New Note" => templates view is displayed
- On app start, click "New Note" => insert from targeted data panel => shortcuts view is displayed
- On app start, click "New Note" => set focus in editor => shortcuts view is displayed
- On app start, click "New Note" => set focus in editor, click clinical notes list, click an existing note, click back on the in progress note => shortcuts view is displayed
- On app start, click "New Note" => click "PLACEHOLDERS", select <staging> => view stays in placeholders view
- on app start, click "New Note" => click  "SHORTCUTS", select @patient => view stays in shortcuts view
- On app start, click "New Note" => insert a template and click ok => shortcuts view is displayed
- On app start, click "New Note" => click on a template, click cancel (don't choose an option) => shortcuts view is displayed



_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._


## Submitter:

**Running the Application**

- [x] Manually tested in Chrome
- [x] Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- N/A Cheat sheet is updated
- N/A Demo script is updated 
- N/A Documentation on Wiki has been updated 
- N/A Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- N/A Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- N/A Added Enzyme-UI tests for slim mode 
- N/A Added Enzyme-UI tests for full mode
- N/A Added backend tests for new functionality added
- N/A Added note parser examples to test new functionality


## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code


## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

jafeltra commented:
This looks good! I tested the cases you listed at the top and they seem good. I did find a bug in a case you didn't have listed, which also breaks some of the tests:

[JA1] When you switch to Shortcuts view and click on a shortcut from the context tray, the context tray switches back to the templates view. Since a number of tests are breaking due to this bug, I can try running them again after.

Also, the updates you made to the test to click on the Shortcuts view seem to be working! However, some of them are still breaking due to JA1. I also couldn't get the tests to run immediately due to an error in the file:

[JA2] There was an error in the tests, which prevented them from running. I commented out the duplicate declaration of sectionItemElements on line 736, to get them to run, but that test still uses the original variable that referenced sectionItemElements, so it still fails and I haven't looked into which are the correct variables to get rid of.

nicoleng12 commented:
Thanks for the comments! I fixed JA1 and JA2. I updated the use cases in the PR description to cover the case that JA1 addresses. @jafeltra please let me know if you find anything else

jafeltra commented:
Looks good! The tests all pass (except for one which is not passing in master) and JA1 and JA2 are fixed. I didn't find anything else!
